### PR TITLE
add config validation in coordinator for MaxActiveUsers

### PR DESCRIPTION
**Pull request by 97amarnathk**
#### Summary
Validate UsersConfig.MaxActiveUsers is not less than clusterConfig.MaxActiveUsers

#### Ticket Link
Github: https://github.com/mattermost/mattermost-server/issues/19676
JIRA: https://mattermost.atlassian.net/browse/MM-42253

### CODE REVIEW

**Review by streamer45 — CHANGES_REQUESTED** [review]
Thanks @97amarnathk, a couple of suggestions:

This wasn't specified in the ticket (sorry about that) but given the coordinator can run multiple agents the check should be against the product of the value and the number of agents available. In this case:

` ltConfig.UsersConfiguration.MaxActiveUsers*len(clusterConfig.Agents)`

Now, given the check is constrained to `LoadAgentClusterConfig` I am thinking we could follow our convention and add a new `IsValid() error` method in `coordinator/cluster/config.go` to be called by the coordinator. 

Let me know if that makes sense.

**Comment by 97amarnathk** [comment]
@streamer45 I have added the change in validation for max active users. 
Since the validation is not constrained only on LoadAgentClusterConfig (also depends on ltConfig), I am passing ltConfig as a param to clusterConfig.IsValid

File: coordinator/coordinator.go
Comment (streamer45) [inline coordinator/coordinator.go:257]: This line can go away I assume.

**Review by streamer45 — COMMENTED** [review]
Thanks @97amarnathk , one more thing but overall looks great.

We should also fix the failing test, just need to set `MaxActiveUsers` in:

https://github.com/mattermost/mattermost-load-test-ng/blob/7ba3f8a72de583e743fedcf1bf6c4e0358d25d44/api/coordinator_test.go#L42

**Comment by 97amarnathk** [comment]
@streamer45 In the coordinator test, I am setting MaxActiveUsers with the default value (1000)

File: api/coordinator_test.go
Comment (streamer45) [inline api/coordinator_test.go:100]: We need to set this above and the value should be 100 since that's what we are using for the agent:

```patch
diff --git a/api/coordinator_test.go b/api/coordinator_test.go
index 187227d..c67f69d 100644
--- a/api/coordinator_test.go
+++ b/api/coordinator_test.go
@@ -40,6 +40,7 @@ func TestCoordinatorAPI(t *testing.T) {
        config.MonitorConfig.Queries[0].Description = "Query"
        config.MonitorConfig.Queries[0].Query = "query"
        config.ClusterConfig.Agents[0].ApiURL = server.URL
+       config.ClusterConfig.MaxActiveUsers = 100
 
        t.Run("create/destroy", func(t *testing.T) {
                data := struct {
@@ -97,7 +98,6 @@ func TestCoordinatorAPI(t *testing.T) {
                }
 
                data.CoordinatorConfig.ClusterConfig.Agents[0].ApiURL = server.URL
-               data.CoordinatorConfig.ClusterConfig.MaxActiveUsers = 1000
 
                obj := e.POST("/create").WithQuery("id", id).WithJSON(data).
                        Expect().Status(http.StatusCreated).
```

By the way you can also run `make test` locally to verify it works.

**Comment by agnivade** [comment]
@97amarnathk - Are you still interesting in working on this?

**Comment by agarciamontoro** [comment]
Picking this where @97amarnathk left it to try and merge it.

**Comment by agnivade** [comment]
ping @streamer45

**Review by agarciamontoro — APPROVED** [review]
This looks good to me now :tada: Let's wait for Claudio (he's on vacation) before we merge this.

**Review by streamer45 — APPROVED** [review]
Looks good :+1:

**Pull request merged**
